salt/_pillar/metalk8s.py

```diff
@@ -94,13 +94,19 @@ def _load_networks(config_data):
     # MetalLB disabled by default
     networks_data["controlPlane"].setdefault("metalLB", {}).setdefault("enabled", False)
 
-    if networks_data["controlPlane"]["metalLB"]["enabled"] and not networks_data[
-        "controlPlane"
-    ].get("ingress", {}).get("ip"):
-        errors.append(
-            "'ip' for 'ingress' in 'controlPlane' network is mandatory when 'metalLB'"
-            "is enabled"
-        )
+    if networks_data["controlPlane"]["metalLB"]["enabled"]:
+        if not networks_data["controlPlane"].get("ingress", {}).get("ip"):
+            errors.append(
+                "'ip' for 'ingress' in 'controlPlane' network is mandatory when "
+                "'metalLB' is enabled"
+            )
+        if len(networks_data["controlPlane"]["cidr"]) > 1:
+            errors.append(
+                "Control Plane Ingress Virtual IP management with MetalLB is "
+                "currently not supported when you have multiple subnet for "
+                "Control Plane network, see "
```

Comment:
Let's keep the focus on what needs to be done, rather than what cannot be done:

```suggestion
                "Enabling 'metalLB' requires a single 'cidr' in "
                "'controlPlane' network, see "
```